autogluon/task/tabular_prediction/predictor.py

```diff
@@ -315,6 +315,42 @@ def fit_summary(self, verbosity=3):
             print("*** End of fit() summary ***")
         return results
 
+    # TODO: Consider adding time_limit option to early stop the feature importance process
+    def feature_importance(self, model=None, dataset=None, features=None, raw=True, silent=False):
+        """
+        Calculates feature importance scores for the given model.
+        Feature scores represent the score reduction the model achieves when the provided data's feature is randomly shuffled among its rows.
+        A score of 0.01 would indicate that the score dropped by 0.01 when the feature was randomly shuffled.
+        The higher the score a feature has, the more important it is to the model's performance.
+        If a feature has a negative score, this means that the feature is likely harmful to the final model, and a model trained with the feature removed would be expected to achieve a better score.
+        Note that calculating feature importance can be a very computationally expensive process, particularly if the model uses hundreds or thousands of features. In many cases, this can take longer than the original model training.
+        To estimate how long `feature_importance(model, dataset, features)` will take, it is roughly the time taken by `predict_proba(dataset, model)` multiplied by the number of features.
```

Comment:
Should consider adding subsample-based importance scores in the future (even using subsample by default for large datasets IMO).

Reply:
Do you think we should be so specific? If we keep it vague in this regard, we can change the internals easier.

Reply:
Added parameter and description